test/integration/controllermanager/managedseedset/managedseedset_test.go

```diff
@@ -0,0 +1,376 @@
+	var (
+		shoot           *gardencorev1beta1.Shoot
+		seed            *gardencorev1beta1.Seed
+		managedSeed     *seedmanagementv1alpha1.ManagedSeed
+		managedSeedSet  *seedmanagementv1alpha1.ManagedSeedSet
+		selector        labels.Selector
+		err             error
+		shootList       *gardencorev1beta1.ShootList
+		seedList        *gardencorev1beta1.SeedList
+		managedSeedList *seedmanagementv1alpha1.ManagedSeedList
+
+		makeReplicaReady = func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, shootList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(shootList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the Shoot as 'successfully created'")
+			patch := client.MergeFrom(shootList.Items[0].DeepCopy())
+			shootList.Items[0].Status = gardencorev1beta1.ShootStatus{
+				ObservedGeneration: shootList.Items[0].GetGeneration(),
+				LastOperation: &gardencorev1beta1.LastOperation{
+					Type:  gardencorev1beta1.LastOperationTypeCreate,
+					State: gardencorev1beta1.LastOperationStateSucceeded,
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, &shootList.Items[0], patch)).To(Succeed())
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, managedSeedList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(managedSeedList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the ManagedSeed condition as SeedRegistered")
+			patch = client.MergeFrom(managedSeedList.Items[0].DeepCopy())
+			managedSeedList.Items[0].Status = seedmanagementv1alpha1.ManagedSeedStatus{
+				ObservedGeneration: managedSeedList.Items[0].GetGeneration(),
+				Conditions: []gardencorev1beta1.Condition{
+					{Type: seedmanagementv1alpha1.ManagedSeedSeedRegistered, Status: gardencorev1beta1.ConditionTrue},
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, &managedSeedList.Items[0], patch)).To(Succeed())
+
+			By("Create Seed manually as ManagedSeed controller is not running in the test")
+			seed.Name = managedSeedList.Items[0].Name
+			Expect(testClient.Create(ctx, seed)).To(Succeed())
+			log.Info("Created Seed for ManagedSeed", "seed", client.ObjectKeyFromObject(seed), "managedSeed", client.ObjectKeyFromObject(&managedSeedList.Items[0]))
+
+			DeferCleanup(func() {
+				By("Delete Seed")
+				Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+				Eventually(func() error {
+					return testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)
+				}).Should(BeNotFoundError())
+			})
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, seedList, client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(seedList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the Seed as Ready")
+			Expect(testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)).To(Succeed())
```

Comment:
Same here, should not be needed?

Reply:
Without making the Seed object ready , the replica status will not become ready. Hence the Seed object is patched to satisfy the Seed ready condition.

Reply:
Yes, that's fine. I'm talking about the `Get` call, though.